### ISSUE TITLE
View of failed pipeline execution is missing the button to go to the edit mode

### ISSUE THREAD

**Issue by jindrichmynarz** [linkedpipes/etl#376]
When you go to the execution view of a failed pipeline it is missing the button to go to the edit mode, which normal execution views have. The only available button is "Close". Is this intentional? 

Tested using the develop version a7d2cb595f26f8d7b5c50900827dadc5f9f980a8.

**Comment by skodapetr** [comment]
@jindrichmynarz What icon does the fail state have? The "edit" button is visible only for "finished" executions - ie. green icon or red "!" in circle.

**Comment by jakubklimek** [comment]
Yes for executions failed e.g. by restarting the executor. There is no debug from for these. For cancelled executions and executions ended in fail reported by components the debug from functionality should be available.

**Comment by jindrichmynarz** [comment]
The pipeline is marked with question mark in red. BTW, it would be useful to have a text (e.g., `title` attribute) that labels the execution state.

The debug view is available for this pipeline. For example, I can view component execution details or intermediate data via FTP. What is the reason behind not enabling to click through to the edit mode?

**Comment by skodapetr** [comment]
In the edit mode, you can use debug from, manipulate with component mapping etc .. If you edit pipeline for running execution, this can be problematic as the state of the component will be changing under your hands. 
We may use snapshot, however we would need to update backend so it can handle "debug from" running execution. 

A solution might be "detach from execution and edit" button. 
What is your use-case for editing a running pipeline?

**Comment by jindrichmynarz** [comment]
The pipeline is not running. I restarted the executor, since the pipeline contained an error. Therefore, I have no use case for editing a running pipeline. The issue is about consistency in views. If you enable to click through the edit view of a failed pipeline, then this should be available in this case too. Perhaps it is a problem of the pipeline being treated as "running" by the front-end even though its execution was stopped (i.e. the executor was killed).

**Comment by skodapetr** [comment]
The "executor killed" is more of "executor lost" state rather then "pipeline failed".  In the second option we know that pipeline finished and is no longer being executed.  

So the issues is to enable "debug from" for "executor lost" state ?

**Comment by jindrichmynarz** [comment]
No, the issue is simply to make the debug views consistent. Either completely disable the debug view of pipeline executions, for which the executor was killed, or add the button to "Edit pipeline" to such views. I'm not sure what conditional logic decides whether this button is shown, but there's probably some problem in it.

**Comment by jakubklimek** [comment]
The monitor does not know that the executor is killed. It simply does not have any info, which in some cases like timeouts may change in a while. For now lets stop discussing this as this is simply work in progress.

**Comment by jindrichmynarz** [comment]
I think you're discussing a different, fundamentally more difficult issue than I do. What I'm concerned in this issue is simply inconsistent templating logic.

Sure, there's a potentially more difficult question if the execution view should be enabled at all for executions that were killed. But I think there's no difficulty in making the views consistent: you can already navigate to the edit view via the pipeline list, so why not navigate directly via the execution view as is possible from other execution views?

**Comment by jakubklimek** [comment]
Because that is not the same thing. Edit from pipeline view has no connection to the executions. Edit from the execution view is connected (data is reused) from the execution from which "Edit" was clicked. And this is not possible for "hard failed or maybe running" pipelines.

**Comment by jindrichmynarz** [comment]
This was not clear to me. I think the usability can be help if the button to edit pipeline is not missing, but simply disabled with an on-hover title explaining that it is disabled because the pipeline execution was killed.
